metagraph/src/graph/alignment/dbg_aligner.hpp

```diff
@@ -33,6 +33,8 @@ class IDBGAligner {
                              const AlignmentCallback &callback) const = 0;
     DBGQueryAlignment align(std::string_view query,
                             bool is_reverse_complement = false) const;
+
+    virtual const DBGAlignerConfig& get_config() const = 0;
```

Comment:
DBGAlignerConfig seems to be specific to the Seed-and-Extend approach. But this seems to be a more abstract class.
`ISeedAndExtendAligner` is only a child of this. So it's an inconsistency...

Reply:
The config has parameters which would be used in almost all aligners (such as match and mismatch scores), along with those used in seed-and-extend approaches.

One way around this would be to have `build_aligner` return `ISeedAndExtendAligner`

Reply:
Yes, or better cast the thing it returns to `ISeedAndExtendAligner`, before checking `min_seed_length` in it's config.
Then you'll keep the generality and will remove the inconsistency.

Reply:
ok, I did it this way